rp2040-hal/src/timer.rs

```diff
@@ -381,3 +381,50 @@ impl_alarm!(Alarm3 {
     int_name: "TIMER_IRQ_3",
     armed_bit_mask: 0b1000
 });
+
+/// Support for RTIC monotonic trait.
+#[cfg(feature = "rtic-monotonic")]
+pub mod monotonic {
+    use super::{Alarm, Instant, Timer};
+    use fugit::ExtU32;
+
+    /// RTIC Monotonic Implementation
+    pub struct Monotonic<A>(pub Timer, A);
+    impl<A: Alarm> Monotonic<A> {
+        /// Creates a new monotonic.
+        pub const fn new(timer: Timer, alarm: A) -> Self {
+            Self(timer, alarm)
+        }
+    }
+    impl<A: Alarm> rtic_monotonic::Monotonic for Monotonic<A> {
+        type Instant = Instant;
+        type Duration = fugit::MicrosDurationU64;
+
+        const DISABLE_INTERRUPT_ON_EMPTY_QUEUE: bool = false;
+
+        fn now(&mut self) -> Instant {
+            self.0.get_counter()
+        }
+
+        fn set_compare(&mut self, instant: Instant) {
+            // The alarm can only trigger up to 2^32 - 1 ticks in the future.
+            // So, if `instant` is more than 2^32 - 2 in the future, we use `max_instant` instead.
+            let max_instant = self.0.get_counter() + 0xFFFF_FFFE.micros();
+            let wake_at = core::cmp::min(instant, max_instant);
+
+            // Cannot fail
+            let _ = self.1.schedule_at(wake_at);
```

Comment:
Why "cannot fail"?
What about AlarmTooSoon?

Reply:
I do not see any detail about TooSoon cased documented, [rp2040-monotonic](https://github.com/korken89/rp2040-monotonic/blob/master/src/lib.rs#L60) treats it as being after overflow, so 🤷 

What do you think it should do?

Reply:
I don't know how rtic uses the monotonic timer. If it can handle additional interrupts, the safest bet would be to trigger one immediately, I guess?
I see two cases where a TooSoon error could occur:
1) Some logic bug in the time calculations. In that case, it could be better to somehow signal the error, but I don't think there is a good way to do this here? If the calculated time was wrong in the first place, triggering the alarm immediately wouldn't make things worse, I guess.
2) The alarm time was correctly calculated as an instant very close in the future, and something delayed processing so `schedule_at` was called at or after that instant. In that case, triggering the interrupt immediately is the best thing to do.

Reply:
That makes me think, if we can detect it to be TooSoon, we could as well manually pend the interupt in the NVIC 🤔 

Or we could:
1. Pause the timer
2. if timestamp > counter => set alarm to timestamp
3. else set alarm = counter
4. resume timer

This'd required to use a critical_section::with context to synchronize the two cores (in case they have an alarm each) as they'd be accessing in write to the pause reg.

Reply:
Pausing the timer sounds like it wouldn't help the accuracy of that timer.
But just pending the interrupt could be a good solution. Or setting the INTF flag? (And clear it again in clear_compare_flag, I guess.)

Reply:
What do you think about the latest changes?